js/src/App.vue

```diff
@@ -44,6 +62,16 @@ export default {
           // If p is set, this is a ping message that only serves to keep the connection open.
           // Break immediately.
           if (res.p !== undefined) {
+            if (timeoutId) {
+              clearTimeout(timeoutId)
+            }
+
+            timeoutId = setTimeout(() => {
+              // we're disconnected, try to connect again
+              console.warn("Disconnected, try to reconnect a few times")
+              this.reconnect()
+            }, 10000)
+            console.debug("Ping")
```

Comment:
Is `Ping` a good message here? Shouldn't it say something like `Started retry loop` or something a bit more descriptive?